### PR TITLE
-download task support for geoip

### PR DESCRIPTION
**Pull request by rajeshLovesToCode**
Signed-off-by: rajeshLovesToCode <rajesh.dharamdasani3021@gmail.com>

### Description
Implemented downlaod maxmind test mmdb files to resource folder
 
### Issues Resolved
GitHub-issue  #253 
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by dlvenable — APPROVED** [review]

File: data-prepper-plugins/geoip-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/databaseenrich/GetGeoLite2DataTest.java [hunk 1]
```diff
@@ -64,8 +64,8 @@ void getGeoDataTest_without_attributes() throws UnknownHostException {
         GeoIPProcessorService.downloadReady = false;
         Map<String, Object> geoData = getGeoLite2Data.getGeoData(inetAddress, attributes, tempFolderPath);
         Assertions.assertNotNull(geoData);
-        assertThat(geoData.get("country_iso_code"), equalTo("US"));
-        assertThat(geoData.get("ip"), equalTo("2001:4860:4860:0:0:0:0:8888"));
+        assertThat(geoData.get("country_iso_code"), equalTo("FR"));
+        assertThat(geoData.get("ip"), equalTo("2a02:ec00:0:0:0:0:0:0"));
```
Comment (kkondaka) [inline data-prepper-plugins/geoip-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/databaseenrich/GetGeoLite2DataTest.java:68]: Nit. It probably better to make this as 
```
assertThat(geoData.get("ip"), equalTo(IP));
```

File: data-prepper-plugins/geoip-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/databaseenrich/GetGeoLite2DataTest.java [hunk 2]
```diff
@@ -88,8 +88,8 @@ void getGeoDataTest_with_attributes() throws UnknownHostException {
         GeoIPProcessorService.downloadReady = false;
         Map<String, Object> geoData = getGeoLite2Data.getGeoData(inetAddress, attributes, tempFolderPath);
         Assertions.assertNotNull(geoData);
-        assertThat(geoData.get("country_name"), equalTo("United States"));
-        assertThat(geoData.get("ip"), equalTo("2001:4860:4860:0:0:0:0:8888"));
+        assertThat(geoData.get("country_name"), equalTo("France"));
+        assertThat(geoData.get("ip"), equalTo("2a02:ec00:0:0:0:0:0:0"));
```
Comment (kkondaka) [inline data-prepper-plugins/geoip-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/databaseenrich/GetGeoLite2DataTest.java:92]: same here.

**Review by kkondaka — APPROVED** [review]

File: data-prepper-plugins/geoip-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/databaseenrich/GetGeoLite2DataTest.java (hunk 1 above)
Comment (rajeshLovesToCode) [reply data-prepper-plugins/geoip-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/databaseenrich/GetGeoLite2DataTest.java]: hi @kkondaka  pr updated with the fix

File: data-prepper-plugins/geoip-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/databaseenrich/GetGeoLite2DataTest.java (hunk 2 above)
Comment (rajeshLovesToCode) [reply data-prepper-plugins/geoip-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/databaseenrich/GetGeoLite2DataTest.java]: hi @kkondaka  pr updated with the fix

**Review by kkondaka — APPROVED** [review]

**Review by dlvenable — APPROVED** [review]

**Pull request merged**